cpp/src/fil/fil.cu

```diff
@@ -135,11 +144,17 @@ struct forest {
     cudaStream_t stream = h.getStream();
     infer(params, stream);
 
+    printf("inference_output before predict flags (THRESHOLD | SIGMOID | AVG ) or RAW: %x\n", (int)output_);
+    output_t inference_output = output_;
+    if(predict_proba)
+        inference_output &= ~output_t::THRESHOLD;
```

Comment:
Not sure if this is needed, as `predict_proba` is passed into `transform_k` separately.